compiler/damlc/daml-compiler/src/DA/Daml/Compiler/Upgrade.hs

```diff
@@ -152,6 +153,21 @@ templateInstances env externPkgId =
   where
     mod = envMod env
 
+choiceInstances :: Env -> LF.PackageId -> [HsDecl GhcPs]
+choiceInstances env externPkgId =
+    [ generateChoiceInstance env externPkgId templateDT choice
+    | template <- NM.elems $ LF.moduleTemplates mod
+    , choice <- NM.elems $ LF.tplChoices template
+    , not (specialArchiveChoice choice)
```

Comment:
Why do we filter out `Archive`? Yes, we have an `archive` method in the `Template` typeclass but we also generate `Choice` instances for it and that matters if you want to use code that is polymorphic in a `Choice` constraint so just calling `archive` isn’t a workaround.

Reply:
no longer filter Archive